fairlearn/utils/_common.py

```diff
@@ -0,0 +1,41 @@
+def _get_soft_predictions(estimator, X, predict_method):
+    """Returns soft predictions of a classifier using either `predict_proba`
+    or `decision_function` methods.
+
+    Parameters
+    ----------
+    estimator : estimator
+        A scikit-learn compatible estimator
+    X : array-like
+        The intput for which the output is desired
+    predict_method : {'auto', 'predict_proba', 'decision_function', 'predict'\
+            }, default='auto'
+        Defines which method of the ``estimator`` is used to get the output
+        values.
+        - 'auto': use one of ``predict_proba``, ``decision_function``, or
+          ``predict``, in that order.
+        - 'predict_proba': use the second column from the putput of
+          `predict_proba`. It is assumed that the second column represents the
+          positive outcome.
+        - 'decision_function': use the raw values given by the
+          `decision_function`.
+        - 'predict': use the hard values reported by the `predict` method.
+          This is equivalent to what is done in [1]_.
+
+    Returns
+    -------
+    predictions : ndarray
```

Comment:
nit: numpy.ndarray ?
There's some way to make this link to the numpy doc (or does just `ndarray` suffice?)

Reply:
Don't think we can link easily to numpy docs, but also note that these are not gonna be rendered (hopefully) on our site, this is a private function.

Reply:
They get rendered as links in the metrics documentation:
https://fairlearn.org/v0.6.2/api_reference/fairlearn.metrics.html
from things like
https://github.com/fairlearn/fairlearn/blob/b9c19250f086ece6ea437ed8dcaef9e7d9d56a16/fairlearn/metrics/_metric_frame.py#L392

Reply:
@riedgar-ms thanks for the link! That required the full `pandas.<class>` rather than `<class>`, though, which makes me think we need `numpy.ndarray`

Reply:
Yes. There are examples of that on the same page.